criu/include/rst_info.h

```diff
@@ -76,11 +77,16 @@ extern struct task_entries *task_entries;
 
 static inline void lock_last_pid(void)
 {
+	/* No locking required if clone3() is available */
+	if (kdat.has_clone3_set_tid)
```

Comment:
We can't do this. last_pid_mutex guarantees that we will not restore threads when there are helper processes (look at call_helper_process), because a helper process can use one of thread pid-s.